optimize.c

```diff
@@ -70,7 +70,9 @@ bool optimize_compress_PTR(INSTRUCTIONS *instructions, int *position, CODE *code
 bool optimize_clear_loop(INSTRUCTIONS *instructions, int *position, CODE *code) {
 	if (globalOptions.optimize < 2) // Only do on optimization level 2 and higher
 		return false;
-
+	if (instructions->size - (*position) < 3) 
```

Comment:
Can you change the `3` to be based on the number of characters being checked?  I.e. `sizeof(clearLoop) - 1`?